include/h2o/http2_internal.h

```diff
@@ -237,7 +237,11 @@ int h2o_http2_update_peer_settings(h2o_http2_settings_t *settings, const uint8_t
 
 /* frames */
 uint8_t *h2o_http2_encode_frame_header(uint8_t *dst, size_t length, uint8_t type, uint8_t flags, int32_t stream_id);
-void h2o_http2_encode_rst_stream_frame(h2o_buffer_t **buf, uint32_t stream_id, int errnum);
+
+#define h2o_http2_encode_rst_stream_frame(buf, stream_id, errnum) \
+    H2O_BUILD_ASSERT(__builtin_constant_p(errnum)?(errnum < 0):0); \
+    h2o_http2_encode_rst_stream_frame_(buf, stream_id, errnum);
+void h2o_http2_encode_rst_stream_frame_(h2o_buffer_t **buf, uint32_t stream_id, int errnum);
```

Comment:
Please rename the function to `h2o_http2__encode_rst_stream_frame`. An extra `_` after the module name is the way we use to mark internal functions.